packages/common/src/core/schemas/internal/subsetSchema.ts

```diff
@@ -6,6 +6,11 @@ import { IConfigurationSchema } from "../types";
  * @param schema
  * @param props
  * @returns
+ *
+ * TODO: this function needs to be updated to include the
```

Comment:
minor, but I'd suggest moving this TODO down to the 3rd step specifically and making it a bit more generic - we don't need to call out that this is a temporary fix...we are still adding a valid enhancement to this function. I'd write _something_ like for the 3rd step:

```ts
// 3. filter schema compositions (anyOf, allOf, oneOf) and
// conditionals (if/then/else) down to the specified properties
// 
// TODO: enhance subset functionality to filter more complex
// compositional schemas (anyOf, oneOf) and non-if/then/else conditional structures
```